src/rabbit_mgmt_stats_gc.erl

```diff
@@ -134,15 +138,50 @@ gc_batch(Rows, Policies, State = #state{gc_next_key = Cont,
                       Now = floor(
                               time_compat:os_system_time(milli_seconds),
                               State),
-                      gc(Key, Table, Policies, Now),
+                      gc(Key, Table, Config, Now),
                       Key
               end,
-    gc_batch(Rows - 1, Policies, State#state{gc_next_key = NewCont}).
+    gc_batch(Rows - 1, Config, State#state{gc_next_key = NewCont}).
 
-gc(Key, Table, Policies, Now) ->
-    Policy = pget(retention_policy(Table), Policies),
+gc(Key, Table, Config, Now) ->
+    case lists:member(Table, ?PROC_STATS_TABLES) of
```

Comment:
Since the moment the process is initialized, the type of GC is known (proc/aggr). Thus, the init should select the callback function to use and run the `lists:member` and the `pget/2` used below, in order to avoid doing the same operations thousands of times during run time. I know it might look silly, but the stats are very heavy and we should decrease as much as we can the number of reductions they perform.